rhp/v4/validation.go

```diff
@@ -170,7 +170,7 @@ func (req *RPCRenewContractRequest) Validate(pk types.PublicKey, tip types.Chain
 	duration := expirationHeight - hp.TipHeight
 	// calculate the minimum allowance required for the contract based on the
 	// host's locked collateral and the contract duration
-	minRenterAllowance := MinRenterAllowance(hp, duration, req.Renewal.Collateral)
+	minRenterAllowance := MinRenterAllowance(hp, req.Renewal.Collateral)
```

Comment:
`nit` we can move up this check `case req.Renewal.Allowance.IsZero():` and it add to the request validation